lib/kubernetes-deploy/kubernetes_resource.rb

```diff
@@ -192,5 +185,61 @@ def pretty_status
     def kubectl
       @kubectl ||= Kubectl.new(namespace: @namespace, context: @context, logger: @logger, log_failure_by_default: false)
     end
+
+    class Event
+      EVENT_SEPARATOR = "ENDEVENT--BEGINEVENT"
+      FIELD_SEPARATOR = "ENDFIELD--BEGINFIELD"
+      FIELDS = %w(
+        .involvedObject.kind
+        .involvedObject.name
+        .count
+        .lastTimestamp
+        .reason
+        .message
+      )
+
+      def self.go_template_for(kind, name)
+        and_conditions = [
+          %[(eq .involvedObject.kind "#{kind}")],
+          %[(eq .involvedObject.name "#{name}")],
+          '(ne .reason "Started")',
+          '(ne .reason "Created")'
+        ]
+        condition_start = "{{if and #{and_conditions.join(' ')}}}"
+        field_part = FIELDS.map { |f| "{{#{f}}}" }.join(%({{print "#{FIELD_SEPARATOR}"}}))
+        %({{range .items}}#{condition_start}#{field_part}{{print "#{EVENT_SEPARATOR}"}}{{end}}{{end}})
+      end
+
+      def self.from_go_template_blob(blob)
```

Comment:
naming nitpick: this name does not reflect the fact that you'll get back a collection of events not just one